capstone/src/main/webapp/scripts/members.js

```diff
@@ -0,0 +1,103 @@
+function loadMembers() {
+  fetch(`/all-group-members`).then(response => response.json()).then((allGroupMembers) => {
+    const allMembers = allGroupMembers;
+    const memberGrid = document.getElementsByClassName("member-grid-container")[0];
+    memberGrid.innerHTML = '';
+    for (let i = 0; i < allMembers.length; i++) {
+      memberGrid.appendChild(createMemberComponents(allMembers[i]));
+    }
+  }).then(() => {
+    addMemberProfileListener();
+  });
```

Comment:
The previous then is not returning a promise, so I think you can just call the addMemberProfileListener() after the for loop

Reply:
I was mainly using it as a separator, but that makes sense. fixed!